src/battle_script_commands.c

```diff
@@ -8866,6 +8865,72 @@ static void Cmd_various(void)
         gFieldStatuses &= ~STATUS_FIELD_TERRAIN_ANY;    // remove the terrain
         TryToRevertMimicry(); // restore the types of Pokémon with Mimicry
         break;
+    case VARIOUS_JUMP_IF_UNDER_200:
+        // If the Pokemon is less than 200 kg, or weighing less than 441 lbs, then Sky Drop will work. Otherwise, it will fail.
+        if (GetPokedexHeightWeight(SpeciesToNationalPokedexNum(gBattleMons[gBattlerTarget].species), 1) < 441)
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+        else
+            gBattlescriptCurrInstr += 7;
+        return;
+    case VARIOUS_SET_SKY_DROP:
+        gStatuses3[gBattlerTarget] |= STATUS3_SKY_DROPPED;
+        gStatuses3[gBattlerTarget] |= STATUS3_ON_AIR;
+        /* skyDropTargets holds the information of who is the attacker and the target of Sky Drop. 
+       This is needed in the case that multiple Pokemon use Sky Drop in the same turn or if
+		   the target of a Sky Drop faints while in the air.*/
+        if (gBattleStruct->skyDropTargets[0] == 0)
+        {
+            // Store the party index of the target and the attacker within their respective parties
+            gBattleStruct->skyDropTargets[0] = gBattlerPartyIndexes[gBattlerAttacker] + 6;
+            gBattleStruct->skyDropTargets[1] = gBattlerPartyIndexes[gBattlerTarget] + 6;
+        }
+        else
+        {
+            gBattleStruct->skyDropTargets[2] = gBattlerPartyIndexes[gBattlerAttacker] + 6;
+            gBattleStruct->skyDropTargets[3] = gBattlerPartyIndexes[gBattlerTarget] + 6;
```

Comment:
I've been thinking... Mon shouldn't be able to switch while Skydropping or being skydropped, so why is it saving `gBattlerPartyIndexes` instead of battlerIds?

Reply:
Because the mon might faint in the air while it is being sky dropped. If battlerIds were used instead, the following error could (and did) occur: It's a trainer double battle. Pokémon A targets Pokémon B with Sky Drop and takes B into the air. Pokémon C targets B with a hit_in_air move, causing Pokémon B to faint. The trainer of B sends out Pokémon D to replace B. If battlerIds are used, Sky Drop will still deal damage to D, even though the original target of the move was B, since B and D would end up having the same battlerId. Using the party index is the only way to be absolutely sure that the target of Sky Drop on the first turn is still the same target on the second turn.

Reply:
Alright, that makes sense.
One more thing before I approve: When using `gBattlerPartyIndexes` why is it using +- 6 when saving and reading? Wouldn't it be simpler to just save and use the party index without that modification?

Reply:
I think it would be much cleaner to initialize `skyDropTargets` as `0xFF` in `BattleStartSetClearData`, and reset the field for `gActiveBattler` to 0xFF in `SwitchInClearSetData` and `FaintClearSetData`. Then you won't need the +- 6 checks which are very difficult to read/comprehend, and you can go ahead and say the attack turn of sky drop fails if the skydroptarget is 0xFF

Reply:
I did the +-6 thing since I had the cleared value for skyDropTargets as 0, but 0 is a valid index for a party index, so I had to offset it by 6 to make sure there was no overlap over the valid values. Switching the cleared value to 0xFF as ghoulslash suggested made that unnecessary and made a lot of things much simpler besides.